File: sample/trigger_event.cpp
#include <event/dispatcher.hpp>
#include <event/trigger.hpp>
#include <iostream>

int main(int argc, char* argv[])
{
	if (argc != 2)
	{
		std::cerr << "Usage: trigger_event <tick_cnt>" << std::endl;
		return -1;
	}

	size_t tick_cnt = atoi(argv[1]);

	ranger::event::dispatcher disp;
	ranger::event::trigger tr(disp, [&tick_cnt] (ranger::event::trigger& tr)
			{
				if (tick_cnt-- > 0)
				{
					std::cout << "touch: " << tick_cnt << std::endl;
					tr.active();
				}
			});
	tr.active();

	return disp.run();
}

File: sample/trigger_event.lua
require "ranger_lua_event"
event = ranger_lua_event

tick_cnt = 10

disp = event.dispatcher()
tr = event.trigger(disp)
tr:set_event_handler(function(self)
	if tick_cnt > 0 then
		tick_cnt = tick_cnt - 1
		print("touch:", tick_cnt)
		tr:active()
	end
end)

tr:active()
disp:run()
